src/Controller/RegistrationController.php

```diff
@@ -65,41 +67,64 @@ public function __construct (
             new OA\Response(response: 200, description: 'Successfully loaded registration form'),
         ]
     )]
-    public function register(Request $request, RegistrationService $registrationService): Response
+    public function register(Request $request): Response
     {
+        
         $user = new User();
         $form = $this->createForm(RegistrationFormType::class, $user);
         $form->handleRequest($request);
 
         if ($form->isSubmitted() && $form->isValid()) {
-            // Get password from the form
             $plainPassword = $form->get('plainPassword')->getData();
-
-            // Handle the profile picture (if any)
             $profilePicture = $form->get('profile_picture')->getData();
 
             try {
-                // Register the user
+                // Registration user
                 $this->registrationService->registerUser($user, $plainPassword, $profilePicture);
                 
-                // Add success message
-                $this->addFlash('success', 'Registration is successful, now you can login!');
-                
-                return $this->redirectToRoute('app_login');
-            } catch (ProfilePictureUploadException $e) {
-                // Handle profile picture upload error
-                $this->addFlash('error', 'Error uploading profile picture: ' . $e->getAdditionalErrorInfo());
+                // Message about successful registration
+                $this->addFlash('success', 'Registration is successful, please verify your email!');
+                return $this->redirectToRoute('registration_success');
+            }  catch (ProfilePictureUploadException $e) {
+                $this->addFlash('error', 'Error uploading profile picture: ' . $e->getMessage());
             } catch (\Exception $e) {
-                // Handle general errors
                 $this->addFlash('error', 'An unexpected error occurred: ' . $e->getMessage());
             }
         }
 
-        // Render the registration form
         return $this->render('registration/register.html.twig', [
             'registrationForm' => $form->createView(),
         ]);
     }
 
+    #[Route('/registration/success', name: 'registration_success')]
+    public function registrationSuccess(): Response
+    {
+        return $this->render('registration/success.html.twig');
+    }
+
+    // New route for verification email
+    #[Route('/users/{userId}/verify/{token}', name: 'verify_email')]
+    public function verifyEmail(int $userId, string $token): Response
+    {
```

Comment:
don't forward user id to verifyUserEmail function, forward user, and use UserValueResolver for getting user